src/CodeCompliance/Domain/Entity/Report.php

```diff
@@ -107,21 +125,85 @@ public function getPath(): string
      */
     public function hasError(): bool
     {
-        if ($this->getViolations()) {
+        if ($this->hasErrorSeverity($this->getViolations())) {
             return true;
         }
 
         foreach ($this->getPackages() as $package) {
-            if ($package->getViolations()) {
+            if ($this->hasErrorSeverity($package->getViolations())) {
                 return true;
             }
             foreach ($package->getFileViolations() as $violations) {
-                if ($violations) {
+                if ($this->hasErrorSeverity($violations)) {
                     return true;
                 }
             }
         }
 
         return false;
     }
+
+    /**
+     * @param array<\SprykerSdk\SdkContracts\Report\Violation\ViolationInterface> $violations
```

Comment:
It is confusing here. According to the changes you will get objects of `\CodeCompliance\Domain\Entity\Violation` here, that will implement `\SprykerSdk\SdkContracts\Report\Violation\ViolationInterface`, but not only it.

Reply:
You need here only `getSeverity` method, and DTO entity implements `ViolationInterface`, so it is also fine here.

Reply:
Added ViolationInterface ReportInterface PackageViolationReportInterface on the Upgrader level